src/js/components/FilterCriteria.jsx

```diff
@@ -91,7 +91,7 @@ class FilterCriteria extends Component {
                 hintText='Filter Criteria'
                 style={style}
                 value={filter.value}
-                onChange={(ev) => onChangeValue(i, ev.target.value)}
+                onChange={(ev, field, data) => onChangeValue(i, ev.target.value, filter.field, fields.data)}
```

Comment:
the third param doesn't need to be here

`onChange={(ev, field) => onChangeValue(i, ev.target.value, filter.field, fields.data)}`